src/data/api/api.py

```diff
@@ -53,16 +53,27 @@ async def time_check(self, update_time: int = 0):
         log.debug("API | Time Check")
         current_time = int(time())
         time_diff = current_time - update_time
-        if time_diff < self.time_delay:
-            log.debug(f"API | Time Check | {time_diff} seconds")
+        log.debug(f"API | Time Check | {time_diff} seconds")
+        if time_diff > self.time_delay:
             return True
         return False
 
     async def update_all(self):
         log.debug("API | Update All")
         responses = await self.fetch_all()
-        for i, (key, value) in enumerate(self.raw_data.items()):
-            self.raw_data[key]["data"] = responses[i]
+        update_needed = await self.time_check(update_time=self.update_time)
+        if update_needed:
+            # Update Needed
+            for i, (key, value) in enumerate(self.raw_data.items()):
+                self.raw_data[key]["data"] = responses[i]
+                self.raw_data[key]["update_time"] = int(time())
+                if key == "updates":
+                    self.raw_data[key]["data"] = await self.format_steam_news(
+                        self.raw_data[key]["data"]["appnews"]["newsitems"]
+                    )
+            self.update_time = int(time())
+            return True
+        return False
```

Comment:
You can skip the 2 return statements and just return `update_needed` at the end?

Reply:
which immediately also conveys what this method returns without having to read the entire thing

Reply:
Those return statements aren't actually needed, they don't signify anything 😂